crypto/asn1/a_d2i_fp.c

```diff
@@ -92,7 +91,7 @@ void *ASN1_item_d2i_fp(const ASN1_ITEM *it, FILE *in, void *x)
 
 #define HEADER_SIZE   8
 #define ASN1_CHUNK_INITIAL_SIZE (16 * 1024)
-static int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
+int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
```

Comment:
Uhmm, so you mean it should be made a public API? That wasn't my intention at all. Would it be useful as a public API? Mind you, I have nothing against that idea, but that feels like a responsibility a bit beyond what I expected, and certainly not something that should glide in as part of this PR. 

Middle ground, I think your proposed tweak of ASN1_d2i_bio has merit, and I have no problem including that here.